### PR TITLE
Feature/improbability network dropdown

### PR DESCRIPTION
**Pull request by mbalazy**

### CODE REVIEW

**Comment by mbalazy** [comment]
https://github.com/BabelFishProtocol/babelfish-dapp/pull/89/commits/710be2b378a3cedbb871d86a28146b0bd78856a5 is just small UI improvement: it changes the way how progress icons appears on the screen. Instead of 'pushing' the label using `display: flex` now they are just `position: absolute` which prevents layout shift.

image

File: src/store/app/app.test.ts [hunk 1]
```diff
@@ -66,4 +75,24 @@ describe('app store', () => {
       expect(currentBlockSelector(store.getState())).toBe(undefined);
     });
   });
+
+  describe('chainsInCurrentNetwork', () => {
```
Comment (mwiniarsk) [inline src/store/app/app.test.ts:79]: maybe add test for `chainId === undefined` and is there any option that chainId isn't in both?

Comment (mbalazy) [reply src/store/app/app.test.ts:79]: no, there is no way that one chainId will exist in both

File: src/store/app/app.selectors.ts [hunk 2]
```diff
@@ -76,16 +82,27 @@ export const unsupportedNetworkSelector = createSelector(
   }
 );
 
+export const chainsInCurrentNetworkSelector = createSelector(
+  chainIdSelector,
+  (chainId) => {
+    if (chainId === undefined) return undefined;
```
Comment (meho99) [inline src/store/app/app.selectors.ts:88]: I think you could return empty array here
```suggestion
    if (chainId === undefined) return [];
```
In this approach the conditional render in component won't be necessary

**Review by michalsidzej — DISMISSED** [review]

File: src/pages/Improbability/Improbability.component.tsx [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
   state,
   coins,
 }: ImprobabilityComponentProps) => {
-  const { setValue, control } = useForm<ImprobabilityFormValues>({
+  const { setValue, control, resetField } = useForm<ImprobabilityFormValues>({
     mode: 'onChange',
     defaultValues: improbabilityDefaultValues,
   });
 
+  const chainOptions = useSelector(chainsInCurrentNetworkSelector);
```
Comment (meho99) [inline src/pages/Improbability/Improbability.component.tsx:28]: move this line to container

**Review by meho99 — APPROVED** [review]

**Pull request merged**